### PR TITLE
[MM-820]: Added testcases for multiple server files

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added testcases for server files
- state.go
- user.go
- ExecuteCommand function of command.go

### CODE REVIEW

File: server/state_test.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+			expectedJustConnect: true,
+			expectError:         false,
+		},
+		{
+			name:                "State length mismatch",
+			state:               "key1_userID1_channelID1",
+			expectedKey:         "",
+			expectedUserID:      "",
+			expectedChannelID:   "",
+			expectedJustConnect: false,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: No need to initialise, check everywhere

File: server/user_test.go [hunk 2]
```diff
@@ -38,3 +43,101 @@ func TestEncryptUserData(t *testing.T) {
...
+			kvSetUserErr:   nil,
+			kvSetRemoteErr: &model.AppError{Message: "some error occurred while saving the remote id"},
+			expectedErr:    "some error occurred while saving the remote id",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			api := &plugintest.API{}
+			p := &Plugin{
```
Comment (raghavaggarwal2308) [inline server/user_test.go]: I think you can directly use p.setAPI

Comment (raghavaggarwal2308) [reply server/user_test.go]: or just p.API

File: server/user_test.go [hunk 3]
```diff
@@ -38,3 +43,101 @@ func TestEncryptUserData(t *testing.T) {
...
+				EncryptionKey: "demo_encrypt_key",
+			})
+
+			dummyInfo := &UserInfo{
+				UserID:   "dummyUserID",
+				RemoteID: "dummyRemoteID",
+			}
+
+			api.On("KVSet", "token_"+dummyInfo.UserID, mock.Anything).Return(tt.kvSetUserErr)
+			api.On("KVSet", "tbyrid_"+dummyInfo.RemoteID, mock.Anything).Return(tt.kvSetRemoteErr)
```
Comment (raghavaggarwal2308) [inline server/user_test.go]: what is the meaning of tbyrid_

File: server/state_test.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+			expectError:         true,
+			expectedErrMsg:      "status mismatch",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			api := &plugintest.API{}
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: run api.asserExpectations

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/user_test.go (hunk 3 above)
Comment (Kshitij-Katiyar) [reply server/user_test.go]: It is a constant in the code 
`tokenKeyByRemoteID = "tbyrid_"`

File: server/user_test.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply server/user_test.go]: not able to do so

File: server/state_test.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		expectedState  string
+		expectedErrMsg string
+	}{
+		{
+			name:          "Store state successful",
+			userID:        "userID1",
+			channelID:     "channelID1",
+			justConnect:   true,
+			returnError:   nil,
+			expectError:   false,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: Cane remove this

File: server/state_test.go [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+			expectedState: "msteamsmeetinguserstate_userID1_channelID1_true",
+		},
+		{
+			name:           "Error occurred while storing state",
+			userID:         "userID2",
+			channelID:      "channelID2",
+			justConnect:    false,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: same

File: server/state_test.go [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+			returnValue:    []byte(""),
+			returnError:    &model.AppError{Message: "error occurred while getting stored state"},
+			expectError:    true,
+			expectedErrMsg: "error occurred while getting stored state",
+		},
+		{
+			name:          "Valid state retrieved",
+			key:           "dummyKey",
+			returnValue:   []byte("dummyState"),
+			returnError:   nil,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: remove this

File: server/state_test.go [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+			expectedState: "dummyState",
+			expectError:   false,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: same

File: server/state_test.go [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+		key            string
+		returnError    error
+		expectError    bool
+		expectedErrMsg string
+	}{
+		{
+			name:        "Delete state successful",
+			key:         "dummyKey",
+			returnError: nil,
+			expectError: false,
```
Comment (raghavaggarwal2308) [inline server/state_test.go]: same

File: server/user_test.go [hunk 10]
```diff
@@ -38,3 +43,104 @@ func TestEncryptUserData(t *testing.T) {
...
+			name:           "Error Saving RemoteID",
+			kvSetUserErr:   nil,
+			kvSetRemoteErr: &model.AppError{Message: "some error occurred while saving the remote id"},
+			expectedErr:    "some error occurred while saving the remote id",
+		},
+		{
+			name:           "User Info stored successfully",
+			kvSetUserErr:   nil,
+			kvSetRemoteErr: nil,
+			expectedErr:    "",
```
Comment (raghavaggarwal2308) [inline server/user_test.go]: same, check everywhere

File: server/user_test.go [hunk 11]
```diff
@@ -38,3 +43,104 @@ func TestEncryptUserData(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			api := &plugintest.API{}
+			p := &Plugin{
+				MattermostPlugin: plugin.MattermostPlugin{
+					API: api,
+				},
+			}
```
Comment (raghavaggarwal2308) [inline server/user_test.go]: We can create common setupPlugin function for all the testcases

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one.

**Pull request closed without merging**